Packs/ExodusIntelligence/Integrations/ExodusVulnerabilityEnrichment/ExodusVulnerabilityEnrichment.py

```diff
@@ -0,0 +1,207 @@
+                    dest[demisto_field].append(value)
+                except KeyError:
+                    dest[demisto_field] = []
+                    dest[demisto_field].append(value)
+            elif key in WANTED_FIELDS:
+                dest[demisto_field] = value
+    return dest
+
+
+def connect():
+    """Establish a connection to Exodus Intelligence API.
+
+    Returns:
+        exodus_Obj: API Client object.
+    """
+    try:
+        exodus = eve.EVEClient(EMAIL, PASSWORD, CLIENT_ENCRYPTION_KEY, url=EVE_URL)
+    except Exception as e:
+        demisto.debug(f"There was a problem connecting to the server: {e}")
+        return None
+    return exodus
+
+
+def fetch_indicators():
+    """Retrieve vulnerability data from Exodus Intelligence.
+
+    Returns:
+        Bool: True/False based on success or failure
+    """
+    score = 0.0
+    indicators = []
+    formatted_list = []
+    min_xi = 0 if MIN_XI == "" else MIN_XI
+    max_xi = 10 if MAX_XI == "" else MAX_XI
+    try:
+        exodus = connect()
+        demisto.debug("Connected to server")
+        recent_vulns = exodus.get_recent_vulns()
+        try:
+            data = recent_vulns["data"]["items"]
+        except KeyError as e:
+            demisto.debug(f"There was an error getting the data {e}")
+        demisto.debug(f"Fetched {len(data)} total vulnerabilities")
+        for item in data:
+            try:
+                cve = item["cves"][0]
+                report_data = {"cve": cve, "identifier": item["identifier"]}
+                if score >= min_xi and score <= max_xi:
+                    report = exodus.get_report(cve)
+                    vulnerability = exodus.get_vuln(cve)
+                    if report["ok"] is True:
+                        report_data = extract_data(report, report_data)
+                    vuln_data = extract_data(vulnerability, report_data)
+                    formatted_list.append(vuln_data)
+            except KeyError as e:
+                demisto.debug(f"There was a problem: {e}")
+    except Exception as e:
+        demisto.debug(f"Something went wrong: {e}")
+        return False
+
+    if len(formatted_list):
+        for items in formatted_list:
+            try:
+                indicator = {
+                    "value": items["identifier"],
+                    "type": "Exodus Intelligence",
+                    "fields": items,
+                }
+                indicators.append(indicator)
+            except KeyError as e:
+                demisto.debug(f"There was a problem creating indicators: {e}")
+        demisto.createIndicators(indicators)
+    return True
+
+
+def reset_data_stream(days):
+    try:
+        exodus = connect()
+        data = exodus.get_recent_vulns(days)
+        return data["data"]["end_ts"]
+    except (ConnectionError, KeyError) as e:
+        demisto.debug(f"There was a problem connecting to the API {e}")
+        return "Unable to reset Exodus Intelligence data stream."
```

Comment:
May I suggest raising the exception, and then catching it in the main when you handle the error? It should probably fail the command.